ts/src/bybit.ts

```diff
@@ -3521,11 +3521,34 @@ export default class bybit extends Exchange {
         if ((clientOrderId !== undefined) && (clientOrderId.length < 1)) {
             clientOrderId = undefined;
         }
+        const avgPrice = this.omitZero (this.safeString (order, 'avgPrice'));
         const rawTimeInForce = this.safeString (order, 'timeInForce');
         const timeInForce = this.parseTimeInForce (rawTimeInForce);
         const stopPrice = this.omitZero (this.safeString (order, 'triggerPrice'));
-        const takeProfitPrice = this.omitZero (this.safeString (order, 'takeProfit'));
-        const stopLossPrice = this.omitZero (this.safeString (order, 'stopLoss'));
+        const reduceOnly = this.safeValue (order, 'reduceOnly');
+        let takeProfitPrice = this.omitZero (this.safeString (order, 'takeProfit'));
+        let stopLossPrice = this.omitZero (this.safeString (order, 'stopLoss'));
+        const triggerDirection = this.safeString (order, 'triggerDirection');
+        const isAscending = (triggerDirection === '1');
+        const isStopOrderType2 = (stopPrice !== undefined) && reduceOnly;
+        if ((stopLossPrice === undefined) && isStopOrderType2) {
```

Comment:
@Dan-krm @xmatthias This logic seems to work fine with open orders , just not sure yet about filled stop orders

```
bybit.fetchOrder (f3968f70-a51f-4544-ad50-1edd1e0a88e8, LTC/USDT:USDT)
{
  info: {
  },
  id: 'f3968f70-a51f-4544-ad50-1edd1e0a88e8',
  clientOrderId: undefined,
  timestamp: 1695899533898,
  datetime: '2023-09-28T11:12:13.898Z',
  lastTradeTimestamp: 1695899533898,
  lastUpdateTimestamp: 1695899533898,
  symbol: 'LTC/USDT:USDT',
  type: 'limit',
  timeInForce: 'GTC',
  postOnly: false,
  reduceOnly: true,
  side: 'buy',
  price: 50,
  stopPrice: 40,
  triggerPrice: 40,
  takeProfitPrice: 40,
  stopLossPrice: undefined,
  amount: 0.1,
  cost: 0,
  average: undefined,
  filled: 0,
  remaining: 0.1,
  status: 'open',
  fee: { cost: 0, currency: 'USDT' },
  trades: [],
  fees: [ { cost: 0, currency: 'USDT' } ]
}
2023-09-28T11:12:41.118Z iteration 1 passed in 1549 ms
```


```
bybit.fetchOrder (1963af79-8713-4822-9e57-2abcf2e11d3d, LTC/USDT:USDT)
{
  },
  id: '1963af79-8713-4822-9e57-2abcf2e11d3d',
  clientOrderId: undefined,
  timestamp: 1695899604560,
  datetime: '2023-09-28T11:13:24.560Z',
  lastTradeTimestamp: 1695899604560,
  lastUpdateTimestamp: 1695899604560,
  symbol: 'LTC/USDT:USDT',
  type: 'limit',
  timeInForce: 'GTC',
  postOnly: false,
  reduceOnly: true,
  side: 'buy',
  price: 50,
  stopPrice: 65,
  triggerPrice: 65,
  takeProfitPrice: undefined,
  stopLossPrice: 65,
  amount: 0.1,
  cost: 0,
  average: undefined,
  filled: 0,
  remaining: 0.1,
  status: 'open',
  fee: { cost: 0, currency: 'USDT' },
  trades: [],
  fees: [ { cost: 0, currency: 'USDT' } ]
```

Reply:
thinking about it a bit further, i don't think the logic will change uppon filling.

it'll still have been a reduceOnly order - and neither side, nor triggerdirection will have changed (though it may be filled or canceled, and have other things set slightly differently.

Reply:
@xmatthias yeah if the order is reduce only I think we can always do this inference, will merge it soon